lib/class-wp-rest-menu-items-controller.php

```diff
@@ -745,13 +667,33 @@ public function get_item_schema() {
 
 		$schema['properties']['title'] = array(
 			'description' => __( 'The title for the object.', 'gutenberg' ),
-			'type'        => 'object',
+			'type'        => array( 'object', 'string' ),
 			'context'     => array( 'view', 'edit', 'embed' ),
 			'arg_options' => array(
-				// Note: sanitization implemented in self::prepare_item_for_database().
-				'sanitize_callback' => null,
-				// Note: validation implemented in self::prepare_item_for_database().
-				'validate_callback' => null,
+				'validate_callback' => static function ( $value, $request ) {
+					if ( 'custom' === $request['type'] && '' === $value ) {
+						return new WP_Error(
+							'rest_title_required',
+							__( 'Title required if menu item of type custom.', 'gutenberg' )
+						);
+					}
```

Comment:
This code will never be executed if the title is missing. I considered the following:
1. Move it over to `type` validate_callback - doesn't seem right, this error is specific to `title`.
2. Set a default value of `""` to ensure this callback is fired on every request. Far from perfection, let's imagine the title is set in the database but is missing from the request, it would override the stored title with an empty string. 
3. Make this field universally required 
4. Add a concept of "conditionally required" parameter to the REST API.
5. Add a controller-wide `validate_callback` and `sanitize_callback` to deal with cases that don't neatly fit into property-specific ones.